packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -11,7 +16,195 @@ declare global {
+    }
+
+    /**
+     * @internal
+     */
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+        this.unbindEditorTransactionEvent();
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonClick(): void {
+        this.tiptapEditor.chain().focus().toggleBold().run();
+    }
+
+    /**
+     * Toggle the bold mark and focus back to the editor
+     * @internal
+     */
+    public boldButtonKeyDown(e: KeyboardEvent): boolean {
+        if (this.isDesiredKeyDownForButton(e)) {
+            this.tiptapEditor.chain().focus().toggleBold().run();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Toggle the italics mark and focus back to the editor
+     * @internal
+     */
+    public italicsButtonClick(): void {
+        this.tiptapEditor.chain().focus().toggleItalic().run();
+    }
+
+    /**
+     * Toggle the italics mark and focus back to the editor
+     * @internal
+     */
+    public italicsButtonKeyDown(e: KeyboardEvent): boolean {
+        if (this.isDesiredKeyDownForButton(e)) {
+            this.tiptapEditor.chain().focus().toggleItalic().run();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Toggle the unordered list node and focus back to the editor
+     * @internal
+     */
+    public bulletListButtonClick(): void {
+        this.tiptapEditor.chain().focus().toggleBulletList().run();
+    }
+
+    /**
+     * Toggle the unordered list node and focus back to the editor
+     * @internal
+     */
+    public bulletListButtonKeyDown(e: KeyboardEvent): boolean {
+        if (this.isDesiredKeyDownForButton(e)) {
+            this.tiptapEditor.chain().focus().toggleBulletList().run();
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Toggle the ordered list node and focus back to the editor
+     * @internal
+     */
+    public numberedListButtonClick(): void {
+        this.tiptapEditor.chain().focus().toggleOrderedList().run();
+    }
+
+    /**
+     * Toggle the ordered list node and focus back to the editor
+     * @internal
+     */
+    public numberedListButtonKeyDown(e: KeyboardEvent): boolean {
+        if (this.isDesiredKeyDownForButton(e)) {
+            this.tiptapEditor.chain().focus().toggleOrderedList().run();
+            return false;
+        }
+        return true;
+    }
+
+    private initializeEditor(): void {
+        if (this.$fastController.isConnected) {
+            const extensions = [
+                /**
+                 * Tiptap starter-kit provides the basic formatting options such as bold, italics, lists etc,. along with some necessary nodes and extensions.
+                 * https://tiptap.dev/api/extensions/starter-kit
+                 * Disabled other not supported marks and nodes for the initial pass.
+                 */
+                StarterKit.configure({
```

Comment:
We could install the individual extensions and push them during the initialization, instead of disabling the extensions in the StarterKit. As mentioned on the [Tiptap's page](https://tiptap.dev/api/nodes/document), some nodes are necessary to be added to the extensions like the `paragraph`, `text`, and `Document` nodes to support these basic features. So I have added all the required nodes to enable the basic support.